### ISSUE TITLE
Magento CE 2.2.2 - No Shipping Method for Orders with VAT ID

### ISSUE THREAD

**Issue by Chiemo** [magento/magento2#12861]
After upgrading to magento CE 2.2.2:
whenever a b2b customer with validated VAT ID tries to put an order there is no shipping method offered in one page checkout although order summary in cart view is showing the correct values, i.e. correct shipping method cost and tax of zero due to shipment abroad with validated VAT ID

### Preconditions
- Upgrade to Magento CE 2.2.2
- MySQL

### Steps to reproduce
1. Register as a new b2b customer and provide all necessary address information including VAT-ID
2. VAT-ID is successfull validated in customer account
3. Customer account correctly assigned to specified customer group with tax class
4. Fixed cost shipping method defined in backend for all countries
5. Store information defined in backend including validated VAT-ID
6. Customer puts product in cart
7. Cart view order summary shows correctly shipping cost and total order cost excluding VAT
8. Clicking checkout button starts magento check out process
9. In the next step "shipping" for some seconds a red block is shown with a "!" icon
10. On the bottom "sorry, no shipping method available" shows up

### Expected result
1.  Shipping method should be available

### Actual result
1. No shipping method available
2. Checkout is not possible

Thanks for your help,

Chiemo

**Comment by magento-engcom-team** [comment]
@Ctucker9233 , thank you for your report.
In your case, you use ["Magento Open Source" or "Magento Commerce 2.2 B2B"](https://magento.com/resources/technical?_ga=2.179857572.1827142047.1506071800-415199849.1506071800)?

**Comment by Chiemo** [comment]
I checked again the reason that no shipping method is offered.

Script stops at:

- jquery.js:10254  POST http://myaddress/rest/default/V1/carts/mine/estimate-shipping-methods-by-address-id 500 (Internal Server Error)

- jquery.js:10254 	xhr.send( ( options.hasContent && options.data ) || null );

Any ideas how to fix this?

Thanx,

Chiemo

**Comment by magento-engcom-team** [comment]
@Chiemo  thank you for your report,
But please first answer question from our last post.

**Comment by Chiemo** [comment]
@magento-engcom-team: I use magento open source ce 2.2.2 version. Was that your question?

Thanx,

Chiemo

**Comment by magento-engcom-team** [comment]
@Chiemo thank you for your answer.
Then please specify what do you mean "b2b customer".

**Comment by Chiemo** [comment]
- b2b customer = business customer who buys as a company  vs. 
- b2c customer = private customer who buys for his own use

The shop I am working on offers products to business customers in europe. The shop is located in Germany. Accoding to EU rules a business customer let's say from France with a valid VAT-ID can buy excluding VAT.

The shop system has to check during checkout if a business customer has a valid VAT-ID and if he needs to pay tax or not. Magento Open Source CE 2.1.X did not have any problems. Business customers with valid VAT-ID could select shipping methods. In Magento Open Source CE 2.2.2 a business customer with valid VAT-ID cannot checkout due to the fact that no shipping method is offered.

Hope this helps for you understanding.

**Comment by Chiemo** [comment]
SUCCESS!

I checked the extensions and found a shipping extension we did not install: **Temando_Shipping**. Disabling this module, deleting the contents of the pub/static folder, deploying the static content again and refreshing the cache solved the problem. Shipping methods are now available for b2b customers with valid VAT-ID and checkout is possible. 

The solution for my problem was already published via issue #12983.

I wonder what the disabled module temando_shipping is meant for.
